src/sentry/overwatch_webhooks/overwatch_consent/impl.py

```diff
@@ -0,0 +1,42 @@
+# Please do not use
+#     from __future__ import annotations
+# in modules such as this one where hybrid cloud data models or service classes are
+# defined, because we want to reflect on type annotations and avoid forward references.
+
+from sentry.constants import ENABLE_PR_REVIEW_TEST_GENERATION_DEFAULT, HIDE_AI_FEATURES_DEFAULT
+from sentry.models.organization import Organization
+from sentry.overwatch_webhooks.overwatch_consent.model import RpcOrganizationConsentStatus
+from sentry.overwatch_webhooks.overwatch_consent.service import OverwatchConsentService
+
+
+class DatabaseBackedOverwatchConsentService(OverwatchConsentService):
+    def get_organization_prevent_consent_status(
+        self, *, organization_ids: list[int], region_name: str
+    ) -> dict[int, RpcOrganizationConsentStatus]:
+        """
+        Get consent status for multiple organizations in a region.
+
+        Consent is determined by the combination of 2 different organization option values:
+        - sentry:hide_ai_features should be False (default)
+        - sentry:enable_pr_review_test_generation should be True (default is False)
+        """
+        organizations = Organization.objects.filter(id__in=organization_ids)
+
+        result: dict[int, RpcOrganizationConsentStatus] = {}
+
+        for org in organizations:
+            hide_ai_features = org.get_option("sentry:hide_ai_features", HIDE_AI_FEATURES_DEFAULT)
+            pr_review_test_generation_enabled = bool(
```

Comment:
nit: shouldn't need to cast to bool?

Reply:
Oop, yeah let me fix that.

Reply:
Err so this could be `None` as well, so I updated both of these to cast to bool instead.